### PR TITLE
Feature/g0 5341 sales purchase register and country support stage

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.ts [hunk 1]
```diff
@@ -54,7 +59,7 @@ export class PurchaseRegisterExpandComponent implements OnInit {
     bsValue = new Date();
 
     constructor(private store: Store<AppState>, private invoiceReceiptActions: InvoiceReceiptActions, private activeRoute: ActivatedRoute, private router: Router, private _cd: ChangeDetectorRef, private _generalActions: GeneralActions) {
-        this.PurchaseRegisteDetailedResponse$ = this.store.pipe(select(p => p.receipt.PurchaseRegisteDetailedResponse), takeUntil(this.destroyed$));
+        this.purchaseRegisteDetailedResponse$ = this.store.pipe(select(p => p.receipt.PurchaseRegisteDetailedResponse), takeUntil(this.destroyed$));
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.ts:62]: rename p

File: apps/web-giddh/src/app/reports/components/report-details-components/report.details.component.ts [hunk 2]
```diff
@@ -293,4 +308,23 @@ export class ReportsDetailsComponent implements OnInit {
...
     }
+
+    /**
+     * Calculates the sales register total
+     *
+     * @private
+     * @param {*} transaction Sales transaction
+     * @memberof ReportsDetailsComponent
+     */
+    private setSalesRegisterTotal(transaction: any): void {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/reports/components/report-details-components/report.details.component.ts:319]: if this code is used in multiple places then you can use this block of code in utility or service

File: apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.ts [hunk 3]
```diff
@@ -23,13 +25,17 @@ export class SalesRegisterExpandComponent implements OnInit {
     public SalesRegisteDetailedItems: SalesRegisteDetailedResponse;
     public from: string;
     public to: string;
-    public SalesRegisteDetailedResponse$: Observable<SalesRegisteDetailedResponse>;
+    public salesRegisteDetailedResponse$: Observable<SalesRegisteDetailedResponse>;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.ts:28]: Add Doc

File: apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.ts [hunk 4]
```diff
@@ -61,7 +67,7 @@ export class SalesRegisterExpandComponent implements OnInit {
     bsValue = new Date();
 
     constructor(private store: Store<AppState>, private invoiceReceiptActions: InvoiceReceiptActions, private activeRoute: ActivatedRoute, private router: Router, private _cd: ChangeDetectorRef, private _generalActions: GeneralActions) {
-        this.SalesRegisteDetailedResponse$ = this.store.pipe(select(p => p.receipt.SalesRegisteDetailedResponse), takeUntil(this.destroyed$));
+        this.salesRegisteDetailedResponse$ = this.store.pipe(select(p => p.receipt.SalesRegisteDetailedResponse), takeUntil(this.destroyed$));
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/reports/components/salesRegister-expand-component/sales.register.expand.component.ts:70]: rename p

File: apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.html [hunk 5]
```diff
@@ -229,14 +227,14 @@
                         </th>
                     </tr>
                 </thead>
-                <tbody *ngIf="(PurchaseRegisteDetailedResponse$ | async).items.length">
+                <tbody *ngIf="(purchaseRegisteDetailedResponse$ | async).items.length">
                     <tr *ngFor="
-                            let item of (PurchaseRegisteDetailedResponse$ | async)
+                            let item of (purchaseRegisteDetailedResponse$ | async)
                                 .items
                         ">
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/reports/components/purchase-register-expand-component/purchase.register.expand.component.html:234]: run formatter

File: apps/web-giddh/src/app/reports/components/purchase-register-component/purchase.register.component.ts [hunk 6]
```diff
@@ -292,4 +306,23 @@ export class PurchaseRegisterComponent implements OnInit {
...
+    /**
+     * Calculates the purchase register total
+     *
+     * @private
+     * @param {*} transaction Purchase transaction
+     * @memberof PurchaseRegisterComponent
+     */
+    private setPurchaseRegisterTotal(transaction: any): void {
+        const item = _.cloneDeep(transaction);
+        this.purchaseRegisterTotal.purchase += item.debitTotal;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/reports/components/purchase-register-component/purchase.register.component.ts:319]: Nullity check item and transaction

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 7]
```diff
@@ -488,14 +493,6 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
             if (!this.shouldShowLoader) {
                 return;
             }
-            // if (stateLoader.show) {
-            //     this.startLoader(true);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:492]: Remove commented code

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 8]
```diff
@@ -488,14 +493,6 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
             if (!this.shouldShowLoader) {
                 return;
             }
-            // if (stateLoader.show) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:491]: Remove commented code

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 9]
```diff
@@ -229,6 +232,11 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
                 this.defaultWarehouse = (warehouseData.defaultWarehouse) ? warehouseData.defaultWarehouse.uniqueName : '';
             }
         });
+        this.store.pipe(select(appState => appState.company), takeUntil(this.destroyed$)).subscribe((companyData: CurrentCompanyState) => {
+            if (companyData) {
+                this.isTcsTdsApplicable = companyData.isTcsTdsApplicable;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:237]: Nullity check isTcsTdsApplicable

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 10]
```diff
@@ -259,6 +262,12 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
             }
         });
 
+        this.store.pipe(select(appState => appState.company), takeUntil(this.destroyed$)).subscribe((companyData: CurrentCompanyState) => {
+            if (companyData) {
+                this.isTcsTdsApplicable = companyData.isTcsTdsApplicable;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:267]: Nullity check isTcsTdsApplicable

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts (hunk 7 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:492]: It is already removed

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts (hunk 8 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:491]: It is already removed @rohit-walkover

File: apps/web-giddh/src/app/reports/components/report-details-components/report.details.component.ts (hunk 2 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/reports/components/report-details-components/report.details.component.ts:319]: Actually the calculation varies based on the module therefore not included in utility service @rohit-walkover

**Pull request merged**